openshift/alerts.yaml

```diff
@@ -0,0 +1,59 @@
+groups:
+- name: obsctl-reloader.rules
+  rules:
+  - alert: ObsCtlRulesStoreInternalServerError
+    annotations:
+      description: Failed to send rules from tenant {{ $labels.tenant }} to store {{ $value | humanizePercentage }}% of the time with a 5xx or 4xx status code.
+      summary: Failing to send rules to Observatorium.
+    expr: |
+      (
+        sum_over_time(obsctl_reloader_prom_rules_store_ops_total{status_code=~"5..|4..", job="obsctl-reloader"}[5m])
+      /
+        sum_over_time(obsctl_reloader_prom_rules_store_ops_total{status_code=~"2..", job="obsctl-reloader"}[5m])
+      ) or vector(0)
+      > 0.10
+    for: 10m
+    labels:
+      severity: critical
+  - alert: ObsCtlRulesSetFailure
+    annotations:
+      description: obsctl-reloader is failing to set rules for tenant {{ $labels.tenant }} in the rules store {{ $value | humanizePercentage }}% of the time due to {{ $labels.reason }}.
+      summary: Failing to set rules in the rules store.
+    expr: |
+      (
+        sum_over_time(obsctl_reloader_prom_rule_set_failures_total{job="obsctl-reloader"}[5m])
+      /
+        sum_over_time(obsctl_reloader_prom_rule_set_total{job="obsctl-reloader"}[5m])
+      ) or vector(0)
+      > 0.10
+    for: 10m
+    labels:
+      severity: warning
+  - alert: ObsCtlRulesStoreAuthenticationError
+    annotations:
+      description: Failed to authenticate tenant {{ $labels.tenant }} with Observatorium.
+      summary: Failing to authenticate tenant with Observatorium.
+    expr: |
+      (
+        sum_over_time(obsctl_reloader_prom_rules_store_ops_total{status_code=~"403", job="obsctl-reloader"}[5m])
+      /
+        sum_over_time(obsctl_reloader_prom_rules_store_ops_total{status_code=~"2..", job="obsctl-reloader"}[5m])
+      ) or vector(0)
+      > 0.10
+    for: 10m
+    labels:
+      severity: warning
```

Comment:
What if multiple tenants are not able to authenticate, would we still keep it at warning?

Reply:
Yes. Unfortunately there's no good solution here that will be automatically good for every setup. It's even not possible to know in advance how many tenants there are in total to know whether it's 10%, 20%, or 100% of the tenants with issues (what would be the threshold to consider it more severe?). This is mostly an informative alert.

Plus authentication verification should live where it's verified, in Observatorium API.